astropy/units/quantity_helper/function_helpers.py

```diff
@@ -561,16 +566,22 @@ def close(a, b, rtol=1e-05, atol=1e-08, *args, **kwargs):
     return (a, b, rtol, atol) + args, kwargs, None, None
 
 
-@function_helper
+@dispatched_function
 def array_equal(a1, a2, equal_nan=False):
-    args, unit = _quantities2arrays(a1, a2)
-    return args, dict(equal_nan=equal_nan), None, None
+    try:
+        args, unit = _quantities2arrays(a1, a2)
+        return np.array_equal(*args, equal_nan=equal_nan), None, None
```

Comment:
This is totally fine as is, but my sense would be to have the `try/except` narrowly around what can raise the error (i.e., `_quantities2arrays`), and thus put the `return` either in an else clause or just at the end.

Reply:
Yeah, I agree that the `try/except` should be more focused, I'll push a quick update.